Service/DoctrineRepositoryFactory.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/*
+ * This file is part of the ElaoAdminBundle.
+ *
+ * (c) 2014 Elao <contact@elao.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Elao\Bundle\AdminBundle\Service;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\Repository\DefaultRepositoryFactory;
+use Doctrine\ORM\Repository\RepositoryFactory;
+use Symfony\Component\DependencyInjection\ContainerAwareInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+final class DoctrineRepositoryFactory implements RepositoryFactory, ContainerAwareInterface
+{
+    /**
+     * Default repository factory
+     *
+     * @var RepositoryFactory
+     */
+    private $defaultFactory;
+
+    /**
+     * Repositories
+     *
+     * @var array
+     */
+    private $repositories = [];
+
+    /**
+     * Constructor
+     */
+    public function __construct()
+    {
+        $this->defaultFactory = new DefaultRepositoryFactory();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setContainer(ContainerInterface $container = null)
```

Comment:
The [`ContainerAwareTrait`](https://github.com/symfony/dependency-injection/blob/master/ContainerAwareTrait.php) trait should be used instead. It would fix the missing `private $container` btw